apiserver/params/cloud.go

```diff
@@ -114,3 +114,67 @@ type CloudSpecResult struct {
 type CloudSpecResults struct {
 	Results []CloudSpecResult `json:"results,omitempty"`
 }
+
+// CloudCredentialRequestParam defines a credential in terms of its cloud and name.
+// It is used to request detailed content for the credential stored on the controller.
+type CloudCredentialRequestParam struct {
+	CloudName      string `json:"cloud-name"`
+	CredentialName string `json:"credential-name"`
+}
+
+func (p CloudCredentialRequestParam) IsEmpty() bool {
+	return p.CloudName == "" && p.CredentialName == ""
+}
+
+// CredentialContentRequestParam defines an input required to make a valid call
+// to get credentials content stored on the controller.
+type CredentialContentRequestParam struct {
+	Credentials    []CloudCredentialRequestParam `json:"credentials,omitempty"`
+	IncludeSecrets bool                          `json:"include-secrets"`
+}
+
+// CloudCredential contains a cloud credential content.
+type CredentialContent struct {
+	// Name is the short name of the credential.
+	Name string `json:"name"`
+
+	// Cloud is the cloud name to which this credential belongs.
+	Cloud string `json:"cloud"`
+
+	// AuthType is the authentication type.
+	AuthType string `json:"auth-type"`
+
+	// Attributes contains non-secret credential values.
+	Attributes map[string]string `json:"attrs,omitempty"`
+
+	// Secrets contains secret credential values.
+	Secrets map[string]string `json:"secrets,omitempty"`
+}
+
+// ModelAccess contains information about user model access.
+type ModelAccess struct {
```

Comment:
maybe this belongs in model.go?

Reply:
Does it? I could add more properties here that would not make it model but more credential specific, say credential.Owner and credential.Tag...